lib/Sema/SemaBounds.cpp

```diff
@@ -936,6 +934,19 @@ namespace {
         return true;
       }
 
+      if (CK == CK_DynamicPtrBounds || CK == CK_AssumePtrBounds) {
+        BoundsExpr *SrcBounds = S.InferRValueBounds(E);
+        Expr *subExpr = E->getSubExpr();
+        BoundsExpr *subExprBounds = S.InferRValueBounds(subExpr);
+
+	if(subExprBounds->isNone()){
```

Comment:
I'm not sure about it. Could you give me more specific? 
The syntax error is handled during generating bounds expression in SemaExpr and e1 expression has no bounds error is generated at this point.